src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -0,0 +1,466 @@
+        the model. Returns bytes or seekable file like object in the
+        format specified in the content_type request header.
+        """
+
+    @abstractmethod
+    def format_response_payload(self, output: bytes) -> str:
+        """Formats the response body according to the output
+        schema of the model. Returns the data type that is
+        received from the response.
+        """
+
+
+class GPT2ContentFormatter(ContentFormatterBase):
+    """Content handler for LLMs from the OSS catalog."""
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps(
+            {
+                "inputs": {"input_string": [ContentFormatterBase.escape_special_characters(prompt)]},
+                "parameters": model_kwargs,
+            }
+        )
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
+        try:
+            return response_json[0]["0"]
+        except KeyError as e:
+            message = f"""Expected the response to fit the following schema:
+                `[
+                    {{
+                        "0": <text>
+                    }}
+                ]`
+                Instead, received {response_json} and access failed at key `{e}`.
+                """
+            raise KeyError(message)
+
+
+class HFContentFormatter(ContentFormatterBase):
+    """Content handler for LLMs from the HuggingFace catalog."""
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps(
+            {
+                "inputs": [ContentFormatterBase.escape_special_characters(prompt)],
+                "parameters": model_kwargs,
+            }
+        )
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
+        try:
+            return response_json[0]["generated_text"]
+        except KeyError as e:
+            message = f"""Expected the response to fit the following schema:
+            `[
+                {{
+                    "generated_text": <text>
+                }}
+            ]`
+            Instead, received {response_json} and access failed at key `{e}`.
+            """
+            raise KeyError(message)
+
+
+class DollyContentFormatter(ContentFormatterBase):
+    """Content handler for the Dolly-v2-12b model"""
+
+    def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+        input_str = json.dumps(
+            {
+                "input_data": {"input_string": [ContentFormatterBase.escape_special_characters(prompt)]},
+                "parameters": model_kwargs,
+            }
+        )
+        return str.encode(input_str)
+
+    def format_response_payload(self, output: bytes) -> str:
+        response_json = json.loads(output)
```

Comment:
Do we have a document on the return response?

Reply:
It's provided for each of the models the users can deploy to MIR